### PR TITLE
[poc] remove recursive parse on shareinfo

### PR DESCRIPTION
**Pull request by ArtificialOwl**
This is just a concept that will avoid recurvise parsing of all folder/files of a share by link (`/shareinfo`)

will returns `hasChildren` if more sub folder are available, but not parsed.
A second request include `&startAt=<sub_folder_id` will returns info about this sub foler within the share.

### CODE REVIEW

File: apps/files_sharing/lib/Controller/ShareInfoController.php [hunk 1]
```diff
@@ -65,7 +65,7 @@ public function __construct(string $appName,
 	 * @param null $dir
 	 * @return JSONResponse
 	 */
-	public function info($t, $password = null, $dir = null) {
+	public function info($t, $password = null, $dir = null, int $st = 0) {
```
Comment (icewind1991) [inline apps/files_sharing/lib/Controller/ShareInfoController.php:68]: please use a more descriptive name for `$st` or add a comment describing what it is

**Comment by icewind1991** [comment]
In general I'm hesitant to patch up the ShareInfo since we no longer use it ourselves with https://github.com/nextcloud/server/pull/32852

File: apps/files_sharing/lib/Controller/ShareInfoController.php [hunk 2]
```diff
@@ -99,30 +106,79 @@ public function info($t, $password = null, $dir = null) {
...
 
 		$data['children'] = [];
 
 		$nodes = $folder->getDirectoryListing();
 		foreach ($nodes as $node) {
-			$data['children'][] = $this->parseNode($node, $permissionMask);
+			if ($recursive) {
+				$data['children'][] = $this->parseNode($node, $permissionMask, false);
+			} else {
+				$data['hasChildren'] = true;
```
[the remark this answers is not in the document]
Comment (ArtificialOwl) [reply apps/files_sharing/lib/Controller/ShareInfoController.php:131]: are we sure about this ? `$folder->getSize()` returns `0` at this point of the process while containing subfolders

File: apps/files_sharing/lib/Controller/ShareInfoController.php (hunk 1 above)
Comment (ArtificialOwl) [reply apps/files_sharing/lib/Controller/ShareInfoController.php:68]: renaming to `$startAt`

File: apps/files_sharing/lib/Controller/ShareInfoController.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (icewind1991) [reply apps/files_sharing/lib/Controller/ShareInfoController.php:131]: Yeah, with only subfolders (or empty files) it would be `0`

**Pull request closed without merging**